src/types/acls.rs

```diff
@@ -0,0 +1,372 @@
+use crate::error::{Error, ParseError};
+use crate::parse::{parse_many_into, MapOrNot};
+use crate::types::UnsolicitedResponse;
+use imap_proto::types::AclRight;
+use imap_proto::Response;
+use ouroboros::self_referencing;
+use std::borrow::Cow;
+use std::collections::HashSet;
+use std::fmt::{Display, Formatter};
+use std::sync::mpsc;
+
+/// enum used for [Session::set_acl] to specify how the ACL is to be modified.
+pub enum AclModifyMode {
+    /// Replace all ACLs on the identifier for the mailbox
+    Replace,
+    /// Add the given ACLs to the identifier for the mailbox
+    Add,
+    /// Remove the given ACLs from the identifier for the mailbox
+    Remove,
+}
+
+/// Helpful wrapper around the ACL rights vector
+/// Used as input for [Session::set_acl] as output in [ListRights], [MyRights], and [AclEntry]
+#[derive(Debug, Eq, PartialEq)]
+pub struct AclRights {
+    pub(crate) data: HashSet<AclRight>,
+}
+
+impl AclRights {
+    /// Returns true if the AclRights has the provided ACL (either as a char or an AclRight enum)
+    pub fn has_right<T: Into<AclRight>>(&self, right: T) -> bool {
```

Comment:
I wonder if we should just call this `has`?

Reply:
I think this is more akin to `HashSet` than `HashMap`. `HashMap::contains_key` has to specifically say "key" since there are also values, but `HashSet` just has `HashSet::contains`, so let's use that!